### DISCUSSION TITLE
ETM_MULTIASSET

### DISCUSSION THREAD

**Discussion by NiftyIsland** [ETM-Standard/etm-standard.github.io#3] [General]
This is a place to discuss the latest version of the [ETM_MULTIASSET extension](https://etm-standard.github.io/ETM_MULTIASSET_v1.0.0) of the ETM standard.

Feel free to propose new ideas, changes or fixes.

Or, just let the community know that we have your support in adopting this extension!

**Comment by NiftyIsland** [comment]
I think there's a real use case here for a single model asset to have multiple LODs included in the metadata. The benefit being twofold: (1) platforms that do not automatically generate LODs now have LODs to work with and (2) higher-fidelity LODs, i.e. not generated by an automatic process. I'm starting some thinking around how this could be handled any would love to hear thoughts from others thinking about this.

**Comment by madjin** [comment]
VRchat is implementing imposters that users can generate, timestamped here: https://youtu.be/5K4mMn_JLpk?t=532
![image](https://user-images.githubusercontent.com/32600939/232256237-731e171d-e091-4ba0-a60d-48ece05e6a90.png)
Seems like they're using some kinda shader

Overall similar stuff to what I outlined in my dev log: https://hackmd.io/@XR/avatarlod

I think sprite sheets are the way to go, its too easy to break avatars with optimization and shaders have lots of complexity, taking a picture from diff angles captures likeness into a single draw call that can be texture atlassed / instanced for good perf

**Comment by madjin** [comment]
## More notes

Good reference: https://github.com/infosia/avatar-asset-pipeline#create-level-of-details-lod


```json
{
  "name":"glb2vrm0_T_pose_optimize",
  "description":"A-pose to T-pose, apply transforms, optimize it and convert glTF binary (.glb) to VRM spec 0.0",
  "pipelines":[
    {
      "name":"gltf_pipeline",
      "description": "Create glTF binary",
      "components":[
        "glb_T_pose",
        "glb_transforms_apply",
        "glb_overrides"
      ]
    },
    {
      "name":"gltfpack_pipeline",
      "description": "Optimize glTF",
      "components":[
        "gltfpack_execute"
      ]
    },
    {
      "name":"gltf_pipeline",
      "description": "Create VRM",
      "components":[
        "glb_z_reverse",
        "glb_jpeg_to_png",
        "vrm0_fix_joint_buffer",
        "vrm0_default_extensions"
      ]
    }
  ]
}
```

### Blender pipeline

- https://blendermarket.com/products/spritehandler
- https://blendermarket.com/products/get-sheet-done

### Webaverse pipeline
https://github.com/webaverse/app/blob/master/avatar-spriter.js
VRM -> Spritesheet pipeline just needs an export option
![image](https://user-images.githubusercontent.com/32600939/232256328-3047a3d7-eddf-41e8-b305-427d7e278ded.png)

### Unity Pipeline

- https://github.com/Error-mdl/retrosprites what vrchat ppl use to make doom sprite avatars

### Requirements
IMO best to have atleast a walk cycle so avatars walking at a distance look more natural

- Avatar file
- Walk animation
- 4-8+ camera positions
